packages/core-modules/contracts/modules/CoreElectionModule.sol

```diff
@@ -180,8 +192,81 @@ contract CoreElectionModule is IElectionModule, ElectionStorage, OwnableMixin {
         electionData.addressVoted[msg.sender] = true;
     }
 
-    function getNomineeVotes(address nominee) external view override returns (uint) {
-        return _electionStore().electionData.nomineeVotes[nominee];
+    function isElectionEvaluated() external view virtual override returns (bool) {
+        return _electionStore().electionData.isElectionEvaluated;
+    }
+
+    function evaluateElectionBatch() external virtual override {
+        if (!isEpochFinished()) {
+            revert EpochNotFinished();
+        }
+
+        if (_electionStore().electionData.isElectionEvaluated) {
+            revert ElectionAlreadyEvaluated();
+        }
+
+        _evaluateElectionBatchBySimpleCounting();
+    }
+
+    function _evaluateElectionBatchBySimpleCounting() internal virtual {
+        ElectionStore storage store = _electionStore();
+        ElectionData storage electionData = store.electionData;
+
+        uint maxBatchSize = 100;
```

Comment:
This function iterate over the candidates to identify the TOP N (where N is the seats count) voted ones. Usually the number of candidates will be limited and manageable, but in some corner case it could be big enough to prevent this function to run due to gas limitations. The **maxBatchSize** limits the quantity of candidates to be processed on every function call.

**maxBatchSize** could be a parameter or external constant. Set to 100 just using some best guest approach.

Reply:
Update: maxBatchSize is now a parameter that can be configured by the owner